src/realm/object-store/sync/app.cpp

```diff
@@ -659,35 +659,40 @@ void App::log_in_with_credentials(
     BsonDocument body = credentials.serialize_as_bson();
     attach_auth_options(body);
 
-    do_request({HttpMethod::post, route, m_request_timeout_ms,
-                get_request_headers(linking_user, RequestTokenType::AccessToken), Bson(body).to_string()},
-               [completion = std::move(completion), credentials, linking_user,
-                self = shared_from_this()](const Response& response) mutable {
-                   if (auto error = AppUtils::check_for_errors(response)) {
-                       self->log_error("App: log_in_with_credentials failed: %1 message: %2",
-                                       response.http_status_code, error->what());
-                       return completion(nullptr, std::move(error));
-                   }
+    // Always update the location on a login to keep the location metadata up to date in cases where
+    // the transport handles the redirection automatically and does not return the redirect response
+    // Since the user is logged out as part of a deployment model change, the user will need to log in
+    // again, but Core will not know that this was driven by an HTTP redirection.
```

Comment:
[nit] I think the first half of this comment is missing a lot of context on why redirects are relevant here. Maybe reword so it leads with some more context?